src/Contacts/include/BipedalLocomotion/ContactDetectors/FixedFootDetector.h

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file FixedFootDetector.h
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTACT_DETECTORS_FIXED_FOOT_DETECTOR_H
+#define BIPEDAL_LOCOMOTION_CONTACT_DETECTORS_FIXED_FOOT_DETECTOR_H
+
+#include <BipedalLocomotion/ContactDetectors/ContactDetector.h>
+#include <BipedalLocomotion/Contacts/ContactPhaseList.h>
+#include <BipedalLocomotion/Contacts/Contact.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+
+#include <memory>
+
+namespace BipedalLocomotion
+{
+namespace Contacts
+{
+
+/**
+ * The FixedFootDetector is a class that can be used to find the fixed foot given a sequence of
+ * contacts.
+ * The logic used to find is the follows:
+ * - In the single support phase the fixed foot is the stance foot
+ * - In the double support phase the fixed foot is the foot in switch off phase, where switch off is
+ *   the transition phase between stance and swing.
+ *
+ * The aforementioned logic is described by the following diagram
+ * @code{.unparsed}
+ * time         0   1   2   3   4   5   6   7   8   9   10  11  12  13  14  15  16  17
+ * L            |+++|---|+++++++++++|---|+++++++++++|---|+++++++++++|---|+++++++++++|
+ * R            |+++++++++++|---|+++++++++++|---|+++++++++++|---|+++++++++++|---|+++|
+ * stance foot  |LLL|RRR|RRR|LLL|LLL|RRR|RRR|LLL|LLL|RRR|RRR|LLL|LLL|RRR|RRR|LLL|LLL|
+ * @endcode
+ * @warning The FixedFootDetector can handle only bipedal locomotion. Furthermore only one foot is
+ * considered as fixed foot
```

Comment:
Let's document this assumption more in detail maybe?

Saying that, the implementation assumes that the contact phase list only expects two contacts data and that at least one of the contact must be active at all times?